### DISCUSSION TITLE
Access Token is retrieved on each call

### DISCUSSION THREAD

**Discussion by rcalv002** [abuzuhri/Amazon-SP-API-CSharp#166] [Q&A]
It looks like the calls to retrieve an access token are made each time the service is used; i understood there was a cache somewhere to store the token and check that instead of requesting a new one each time.

Since this is a long lived scenario we have registered the service as singleton and the AmazonConnection is only made once in the constructor of this singleton. Is there some configuration missing?

![image](https://user-images.githubusercontent.com/25310133/164565500-4f7d9485-ab43-49da-b764-4a205b04b223.png)

**Comment by abuzuhri** [comment]
HI @rcalv002 

This should not happen unless you change the setting for IsActiveLimitRate to false
Do you do any change ?
```Csharp
var amazonConnection = new AmazonConnection(new AmazonCredential()
{
      .
      .
      IsActiveLimitRate=false
});
```
